perfkitbenchmarker/benchmarks/fio_benchmark.py

```diff
@@ -317,59 +396,38 @@ def Prepare(benchmark_spec):
 
   """
 
-  if FLAGS.fio_jobfile:
-    ignored_flags = {'--' + flag_name
-                     for flag_name in FLAGS_IGNORED_FOR_CUSTOM_JOBFILE
-                     if FLAGS[flag_name].present}
-
-    if ignored_flags:
-      logging.warning('Fio job file specified. Ignoring options "%s"',
-                      ', '.join(ignored_flags))
-
-  if FLAGS.run_for_minutes % MINUTES_PER_JOB != 0:
-    logging.warning('Runtime %s will be rounded up to the next multiple of %s '
-                    'minutes.', FLAGS.run_for_minutes, MINUTES_PER_JOB)
+  WarnOnBadFlags()
 
   vm = benchmark_spec.vms[0]
   logging.info('FIO prepare on %s', vm)
   vm.Install('fio')
 
-  # Fill the disk or file we're using
+  # Choose a disk or file name and optionally fill it
   disk = vm.scratch_disks[0]
-  device_path = disk.GetDevicePath()
   mount_point = disk.mount_point
-  if FLAGS.against_device:
-    logging.info('Umount scratch disk on %s at %s', vm, mount_point)
-    vm.RemoteCommand('sudo umount %s' % mount_point)
 
-  if FLAGS.device_fill_size is not '0':
-    if FLAGS.against_device:
-      fill_path = device_path
-      fill_size = FLAGS.device_fill_size
-    else:
-      fill_path = posixpath.join(mount_point, DEFAULT_TEMP_FILE_NAME)
-      if FLAGS['device_fill_size'].present:
-        fill_size = FLAGS.device_fill_size
-      else:
-        # Default to 90% of capacity because the file system will add
-        # some overhead.
-        fill_size = str(int(DISK_USABLE_SPACE_FRACTION *
-                            1000 * disk.disk_size)) + 'M'
-
-    logging.info('Fill file %s on %s', fill_path, vm)
-    command = GenerateFillCommand(fio.FIO_PATH,
-                                  fill_path,
-                                  fill_size)
-    vm.RemoteCommand(command)
+  logging.info('Umount scratch disk on %s at %s', vm, mount_point)
+  vm.RemoteCommand('sudo umount %s' % mount_point)
 
+  if FLAGS.device_fill_size is not '0':
+    logging.info('Fill device %s on %s', disk.GetDevicePath(), vm)
+    FillDevice(fio.FIO_PATH, vm, disk, FLAGS.device_fill_size)
+
+  if not FLAGS.against_device:
+    vm.FormatDisk(disk.GetDevicePath())
+    # Don't use vm.MountDisk() because that also creates the mount point
+    # and changes its permissions, which we have already done.
+    vm.RemoteCommand('sudo mount %s %s' % (disk.GetDevicePath(), mount_point))
+
+  job_file_string = GetOrGenerateJobFileString(FLAGS.fio_jobfile,
+                                               FLAGS.generate_scenarios,
+                                               FLAGS.against_device,
+                                               disk,
+                                               FLAGS.io_depths,
+                                               FLAGS.working_set_size)
   job_file_path = vm_util.PrependTempDir(LOCAL_JOB_FILE_NAME)
```

Comment:
I can eliminate the extra job file generation by loading the job file from the LOCAL_JOB_FILE_PATH in the Run() method. I thought about doing that, but I thought it might actually be faster to generate again since we wouldn't touch the disk. However, probably simplicity is more important than speed here. What do you think of this solution?

As for changing the job file every run, I see that it might be useful, but I don't know how we would use it right now. Maybe a future feature?